CHANGES.rst

```diff
@@ -9,6 +9,8 @@
     [GH-1289]
   - ``rust-cargo`` does not use the variable ``flycheck-rust-args`` anymore
     [GH-1289]
+  - Improve detection of default directory for ``haskell-ghc`` to consider
+    ``hpack`` project files [GH-1435]
```

Comment:
I'd possibly phrase this as "Consider `hpack` project files when selecting the directory to run  haskell-ghc from.  But I like your phrasing too.